src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -349,5 +411,157 @@ internal static bool TryParseSeverity(string value, out ReportDiagnostic severit
             severity = default;
             return false;
         }
+
+        internal static bool IsPartialGlobalConfig(AnalyzerConfig config) => config.GlobalSection.Properties.ContainsKey(GlobalKey);
```

Comment:
This looks like it's the wrong place, probably should be on the AnalyzerConfig type itself.

Reply:
Same as above. Happy to move it, but doesn't it make more sense that the thing responsible for combining, knows whether or not to combine?

---
In reply to: [423392151](https://github.com/dotnet/roslyn/pull/43889#discussion_r423392151) [](ancestors = 423392151)